var/spack/repos/builtin/packages/umap/test/umaptest.cpp

```diff
@@ -0,0 +1,187 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright 2017-2021 Lawrence Livermore National Security, LLC and other
+// UMAP Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: LGPL-2.1-only
+//////////////////////////////////////////////////////////////////////////////
```

Comment:
If this is getting committed to Spack, it should be MIT/Apache-2.0 and use the required license headers for Spack.  Otherwise we're not a pure MIT/Apache-2.0 project.  Using this (LGPL-2.1) is actually incompatible with Apache-2.0 so we can't accept it this way.

I *think* in this case, @egreen77, @mayagokhale, and @IBPeng are maintainers of `umap`, so it would be better to do a point release of `umap`, include this file with the `umap` release, and not include it with Spack -- then you can more easily modify the test as versions of `umap` progress.  You can make the test conditional by decorating the test method with, e.g., `@when("@2.1.1:")`, and it'll only test when it's at that version.

Reply:
Thanks for pointing this out. After some discussion, it was decided to re-license the test code with Spack's license header.

Reply:
@tgamblin Can you review when you get a chance, please?